module/jobqueue/sealed_header_reader.go

```diff
@@ -26,33 +25,23 @@ func NewSealedBlockHeaderReader(state protocol.State, headers storage.Headers) *
 // AtIndex returns the block header job at the given index.
 // The block header job at an index is just the finalized block header at that index (i.e., height).
 func (r SealedBlockHeaderReader) AtIndex(index uint64) (module.Job, error) {
-	header, err := r.blockByHeight(index)
-	if err != nil {
-		return nil, fmt.Errorf("could not get block by index %v: %w", index, err)
-	}
-
 	sealed, err := r.Head()
 	if err != nil {
 		return nil, fmt.Errorf("could not get last sealed block height: %w", err)
 	}
 
 	if index > sealed {
 		// return not found error to indicate there is no job available at this height
-		return nil, fmt.Errorf("block at index %v is not sealed: %w", index, storage.ErrNotFound)
+		return nil, fmt.Errorf("block at index %d is not sealed: %w", index, storage.ErrNotFound)
 	}
 
-	// the block at height index is sealed
-	return BlockHeaderToJob(header), nil
-}
-
-// blockByHeight returns the block at the given height.
-func (r SealedBlockHeaderReader) blockByHeight(height uint64) (*flow.Header, error) {
-	block, err := r.headers.ByHeight(height)
+	header, err := r.headers.ByHeight(index)
```

Comment:
`SealedBlockHeaderReader` only need to depend on protocol state, which internally manages the headers storage module.

```
header, err := r.state.AtHeight(height).Head()
```

Reply:
I originally used this, but @AlexHentschel asked me to switch it back to take advantage of caching in the headers storage module